node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/RetiredExpirer.java

```diff
@@ -93,7 +95,32 @@ private boolean canRemove(Node node) {
             return false;
         }
 
-        if (node.history().hasEventBefore(History.Event.Type.retired, clock().instant().minus(retiredExpiry))) {
+        if (node.type().isConfigServerLike()) {
+            // Avoid eventual expiry of configserver-like nodes
+
+            if (activeNodes.nodeType(node.type()).asSet().size() < NUM_CONFIG_SERVERS) {
```

Comment:
Nit: `NodeList` has `size` so need to copy into a set.

Reply:
Fixed